### PR TITLE
breaking: Remove content_type and accept parameters from Predictor

### PR DESCRIPTION
**Pull request by bveeramani**
Need to add feature in separate PR before merging this PR.

*Issue #, if available:*

*Description of changes:*
* Remove `content_type` and `accept` from `Predictor` constructor
* Remove `content_type` and `accept` from `create_model` methods
* Set default Predictor `serializer` to `IdentitySerializer()`
* Set default Predictor `deserializer` to `BytesDeserializer()`
* Added `IdentitySerializer()`

*Testing done:*
* Fixed relevant tests

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [ ] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [ ] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: doc/v2.rst [hunk 1]
```diff
@@ -94,6 +94,18 @@ Please instantiate the objects instead.
 The ``update_endpoint`` argument in ``deploy()`` methods for estimators and models has been deprecated.
 Please use :func:`sagemaker.predictor.Predictor.update_endpoint` instead.
 
+``content_type`` and ``accept`` in the Predictor Constructor
+------------------------------------------------------------
+
+The ``content_type`` and ``accept`` parameters have been removed from the
+following methods:
+- ``sagemaker.predictor.Predictor.__init__``
```
Comment (laurenyu) [inline doc/v2.rst:102]: instead of writing out `__init__` I might just say "the following methods and classes"

File: src/sagemaker/algorithm.py [hunk 2]
```diff
@@ -251,37 +253,29 @@ def create_model(
...
-                data and the response content type, and return a sequence of
-                bytes. May provide a content_type attribute that defines the
-                endpoint response Accept content type.
-            content_type (str): The invocation ContentType, overriding any
-                content_type from the serializer
-            accept (str): The invocation Accept, overriding any accept from the
-                deserializer.
+            serializer (sagemaker.serializers.BaseSerializer): A serializer
+                object, used to encode data for an inference endpoint
+                (default: ``IdentitySerializer()``).
```
Comment (laurenyu) [inline src/sagemaker/algorithm.py:275]: you can link directly to the class with

```
:class:`~sagemaker.serializers.IdentitySerializer`
```

(may have misplaced the tilde lol, but the syntax is something like that)

File: doc/v2.rst (hunk 1 above)
Comment (bveeramani) [reply doc/v2.rst:102]: Fixed

File: src/sagemaker/algorithm.py (hunk 2 above)
Comment (bveeramani) [reply src/sagemaker/algorithm.py]: Fixed

File: tests/integ/test_byo_estimator.py [hunk 3]
```diff
@@ -91,6 +91,7 @@ def test_byo_estimator(sagemaker_session, region, cpu_instance_type, training_se
         predictor = model.deploy(1, cpu_instance_type, endpoint_name=job_name)
         predictor.serializer = _FactorizationMachineSerializer()
         predictor.deserializer = sagemaker.deserializers.JSONDeserializer()
+        predictor.content_type = "application/json"
```
Comment (bveeramani) [inline tests/integ/test_byo_estimator.py:94]: In my next PR, I'm going to add serializer and deserializer parameters to `deploy`.

File: tests/integ/test_byo_estimator.py [hunk 4]
```diff
@@ -136,6 +137,7 @@ def test_async_byo_estimator(sagemaker_session, region, cpu_instance_type, train
         predictor = model.deploy(1, cpu_instance_type, endpoint_name=endpoint_name)
         predictor.serializer = _FactorizationMachineSerializer()
         predictor.deserializer = sagemaker.deserializers.JSONDeserializer()
+        predictor.content_type = "application/json"
```
Comment (bveeramani) [inline tests/integ/test_byo_estimator.py:140]: In my next PR, I'm going to add serializer and deserializer parameters to `deploy`.

File: tests/integ/test_tuner.py [hunk 5]
```diff
@@ -903,6 +903,7 @@ def test_tuning_byo_estimator(sagemaker_session, cpu_instance_type):
         predictor = tuner.deploy(1, cpu_instance_type, endpoint_name=best_training_job)
         predictor.serializer = _FactorizationMachineSerializer()
         predictor.deserializer = JSONDeserializer()
+        predictor.content_type = "application/json"
```
Comment (bveeramani) [inline tests/integ/test_tuner.py:906]: In my next PR, I'm going to add serializer and deserializer parameters to `deploy`.

File: tests/integ/test_tuner_multi_algo.py [hunk 6]
```diff
@@ -190,6 +190,7 @@ def _deploy_and_predict(sagemaker_session, tuner, data_set, cpu_instance_type):
             )
         )
         predictor = tuner.deploy(1, cpu_instance_type, endpoint_name=best_training_job)
+        predictor.content_type = "application/json"
```
Comment (bveeramani) [inline tests/integ/test_tuner_multi_algo.py:193]: In my next PR, I'm going to add serializer and deserializer parameters to `deploy`. In that PR, I'll remove `predictor.content_type = ...`.

**Pull request merged**